src/lib.rs

```diff
@@ -105,3 +105,124 @@ mod url_tests {
         assert!(result.is_ok());
     }
 }
+
+#[cfg(test)]
+mod management_tests {
+    use crate::{
+        management::list_files::{ListFiles, Operator, Options, SearchQuery, SearchQueryBuilder},
+        ImageKit,
+    };
+
+    #[tokio::test]
+    async fn list_files() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let result = imagekit.list_files(Options::default()).await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn list_files_with_raw_query_string_search() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let query = SearchQueryBuilder::raw_query_string("name=\"default-image.jpg\"").build();
+        let options = Options::new().search_query(query);
+        let result = imagekit.list_files(options).await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn list_files_without_builder() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let query = SearchQuery("name=\"default-image.jpg\"".into());
+        let options = Options::new().search_query(query);
+        let result = imagekit.list_files(options).await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn list_files_with_name_search() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let query = SearchQueryBuilder::name(Operator::EqualTo, "\"default-image.jpg\"").build();
+        let options = Options::new().search_query(query);
+        let result = imagekit.list_files(options).await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn list_files_with_advanced_search() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let search = SearchQueryBuilder::size(Operator::GreaterThan, 100)
+            .and(SearchQueryBuilder::tags(Operator::In, &["summer-sale"]).build())
+            .build();
+        let options = Options::new().search_query(search);
+        let result = imagekit.list_files(options).await;
+        assert!(result.is_ok());
+    }
```

Comment:
Can we have assertions on resulting string?

Reply:
There are separate tests for the builder where the assertions on the resulting string is made. I just didn't want to pollute this check to be done, where the key objective is to list_files.